shared/src/store/action-reducers/utils/get-element.ts

```diff
@@ -47,3 +51,41 @@ export function getElementByPredicate<
     }
     return element;
 }
+
+export function getBehaviorById(
+    state: Mutable<ExerciseState>,
+    simulatedRegionId: UUID,
+    behaviorId: UUID
+): Mutable<ExerciseSimulationBehaviorState> {
+    const simulatedRegion = getElement(
+        state,
+        'simulatedRegion',
+        simulatedRegionId
+    );
+    const behavior = simulatedRegion.behaviors.find((b) => b.id === behaviorId);
+    if (!behavior) {
+        throw new ReducerError(
+            `Behavior with id ${behaviorId} does not exist in simulated region ${simulatedRegionId}`
+        );
+    }
+    return behavior;
+}
+
+export function getActivityById(
+    state: Mutable<ExerciseState>,
+    simulatedRegionId: UUID,
+    activityId: UUID
+): Mutable<ExerciseSimulationActivityState> {
```

Comment:
Same suggestion regarding the typings as for `getBehaviorById`.

Reply:
Done in 5eecd7a6d36ec59a29d7ba730b569c1102e7bdde